### ISSUE TITLE
[REVIEW]: YoungTab

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#221]
**Submitting author:** @luk87 (<a href="http://orcid.org/0000-0001-9033-1345">Luca Romano</a>)
**Repository:** <a href="https://github.com/luk87/YoungTab" target ="_blank">https://github.com/luk87/YoungTab</a>
**Version:** v1.0
**Editor:** @danielskatz
**Reviewer:** @jmichel7
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.572276" target="_blank">10.5281/zenodo.572276</a>

## Status

[![status](http://joss.theoj.org/papers/024cd1ede1c5936b8be98f85eefd8969/status.svg)](http://joss.theoj.org/papers/024cd1ede1c5936b8be98f85eefd8969)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/024cd1ede1c5936b8be98f85eefd8969"><img src="http://joss.theoj.org/papers/024cd1ede1c5936b8be98f85eefd8969/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/024cd1ede1c5936b8be98f85eefd8969/status.svg)](http://joss.theoj.org/papers/024cd1ede1c5936b8be98f85eefd8969)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/luk87/YoungTab" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/luk87/YoungTab">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v1.0)?
- [x] **Authorship:** Has the submitting author (luk87) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS. @jmichel7 it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As as reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all JOSS reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by danielskatz** [comment]
Reviewer instructions

> Please work through the checklist at the start of this issue.
> 
> If you need any further guidance/clarification take a look at the reviewer guidelines here http://joss.theoj.org/about#reviewer_guidelines
> 
> Please make a publication recommendation at the end of your review
> 
> Any questions, please ask for help by commenting on this issue! 🚀

In brief:

@jmichel7: your job now is to go through the checklist at top of the issue, and check off the items that are good.

If you have specific detailed issues with the submission/software, you can open an issue in the software repository (https://github.com/luk87/YoungTab, select Issues tab, click on the green "New issue" button) and put a comment here in this issue that refer to the URL of that issue.

If you have general issues about this as a paper, feel free to add comments here.

In either case, @luk87 (the submitting author) will be notified of the issues and then can respond to them.  But I will only be notified of comments here.

Once everything is done, add a comment saying so and making a recommendation for what to do next.

Thanks!!

**Comment by danielskatz** [comment]
@jmichel7 - what's going on with this review?

**Comment by jmichel7** [comment]
I found the following issues within the reviewer questions:
- Version number: no version number is visible anywhere within the package.
- Example usage: there is none. However, the graphic interface is easily discoverable.
- Functionality documentation: there is none. It seems from the interface that computation is limited
 to groups of rank 10 or less. Such a limitation leaves the program useful to physicists, but makes it probably useless for mathematicians. Thus the target audience can only be physicists. Also, one would like to know with certainty that there can be no overflows in the computations (are large integers supported?)
- Automated tests: there are none
-Community guidelines: there are none either.

Altogether, the program is just a (rather pleasant) widget doing one very specific computation.
Its functionality is thus very limited. Writing it in C (the source does not contain a single comment) makes it very difficult for other people to modify/extend, but this choice of language was probably guided by the chosen graphical interface.

To summarize: a nice widget but without any future. A similar project with a future would choose another
implementation language and perhaps be part of a more general project for Lie groups and algebras, like the Lie package of Arjeh Cohen and al.

**Comment by jmichel7** [comment]
On Thu, Apr 27, 2017 at 01:39:14PM -0700, Daniel S. Katz wrote:
> @jmichel7 - what's going on with this review?

Thank you for reminding me. I put some comments. Tell me if more is needed in
a review.

Best regards,
------------------------------------------------------------------------------
Jean MICHEL, Groupes et representations, IMJ-PRG UMR7586 tel.(33)157279144
Bureau 639 Bat. Sophie Germain Case 7012 - 75205 PARIS Cedex 13

**Comment by danielskatz** [comment]
Thanks @jmichel7 

@luk87, over to you to respond

**Comment by luk87** [comment]
@jmichel7 and @danielskatz thank you. I try to clarify each point.

> I found the following issues within the reviewer questions:
> 
>     Version number: no version number is visible anywhere within the package.
>     Example usage: there is none. However, the graphic interface is easily discoverable.

I can fix these issues immediately describing an example in the documentation. For the version number I put it in the release as required by GitHub. I will report it in the documentation if needed.

>     Functionality documentation: there is none. It seems from the interface that computation is limited
>     to groups of rank 10 or less. Such a limitation leaves the program useful to physicists, but makes it probably useless for mathematicians. Thus the target audience can only be physicists. Also, one would like to know with certainty that there can be no overflows in the computations (are large integers supported?)
>     Automated tests: there are none

I am going to integrate a control sequence to signal overflow. The overflow could appear in the computation of r and h in the code. I have defined these as floats. The limitation to rank 10 was imposed thinking to the usual problems it was supposed to solve. It eventually could be relaxed and also the computational power could be increased (clearly depending on how far one wants to go it requires different approaches). One way to do this is to look for simplification directly in the matrices r and h. 

>     Community guidelines: there are none either.
> 
> Altogether, the program is just a (rather pleasant) widget doing one very specific computation.
> Its functionality is thus very limited. Writing it in C (the source does not contain a single comment) makes it very difficult for other people to modify/extend, but this choice of language was probably guided by the chosen graphical interface.
> 
> To summarize: a nice widget but without any future. A similar project with a future would choose another
> implementation language and perhaps be part of a more general project for Lie groups and algebras, like the Lie package of Arjeh Cohen and al.

The program is thought just as a simple tool, that can be useful also to people not working specifically on this subject and need to compute the dimension of tensors carrying indices of the classical Lie algebras, for example people working in gauge theory. Even if it was not built as part of a bigger project there could be some developments, as the inclusion of the computation of product of representations and the possibility to do branching of representations.

**Comment by luk87** [comment]
@jmichel7 and @danielskatz thank you. I try to clarify each point.

> I found the following issues within the reviewer questions:
> 
>     Version number: no version number is visible anywhere within the package.
>     Example usage: there is none. However, the graphic interface is easily discoverable.

I can fix these issues immediately describing an example in the documentation. For the version number I put it in the release as required by GitHub. I will report it in the documentation if needed.

>     Functionality documentation: there is none. It seems from the interface that computation is limited
>     to groups of rank 10 or less. Such a limitation leaves the program useful to physicists, but makes it probably useless for mathematicians. Thus the target audience can only be physicists. Also, one would like to know with certainty that there can be no overflows in the computations (are large integers supported?)
>     Automated tests: there are none

I am going to integrate a control sequence to signal overflow. The overflow could appear in the computation of r and h in the code. I have defined these as floats. The limitation to rank 10 was imposed thinking to the usual problems it was supposed to solve. It eventually could be relaxed and also the computational power could be increased (clearly depending on how far one wants to go it requires different approaches). One way to do this is to look for simplification directly in the matrices r and h. 

>     Community guidelines: there are none either.
> 
> Altogether, the program is just a (rather pleasant) widget doing one very specific computation.
> Its functionality is thus very limited. Writing it in C (the source does not contain a single comment) makes it very difficult for other people to modify/extend, but this choice of language was probably guided by the chosen graphical interface.
> 
> To summarize: a nice widget but without any future. A similar project with a future would choose another
> implementation language and perhaps be part of a more general project for Lie groups and algebras, like the Lie package of Arjeh Cohen and al.

The program is thought just as a simple tool, that can be useful also to people not working specifically on this subject and need to compute the dimension of tensors carrying indices of the classical Lie algebras, for example people working in gauge theory. Even if it was not built as part of a bigger project there could be some developments, as the inclusion of the computation of product of representations and the possibility to do branching of representations.

**Comment by danielskatz** [comment]
@luk87 Please let us know when you have made these changes.

@arfon Can you comment on the issue of scope/weight:
> Altogether, the program is just a (rather pleasant) widget doing one very specific computation.
> Its functionality is thus very limited. Writing it in C (the source does not contain a single comment) makes it very difficult for other people to modify/extend, but this choice of language was probably guided by the chosen graphical interface.
> 
> To summarize: a nice widget but without any future. A similar project with a future would choose another
> implementation language and perhaps be part of a more general project for Lie groups and algebras, like the Lie package of Arjeh Cohen and al.

**Comment by luk87** [comment]
@danielskatz  yes of course. I have already uploaded the new code with the overflow control and I will try to fix the other issues as soon as possible.

**Comment by arfon** [comment]
> @arfon Can you comment on the issue of scope/weight:

@danielskatz - I think the (limited) scope is fine. Sounds like @jmichel7 is recommending it might be worth adding some documentation to the code to improve the legibility?

**Comment by luk87** [comment]
@danielskatz I have released the software with the overflow control and I have added to the documentation and example and a test to verify functionality. Furthermore I have added comments for a better reading of the source files, main20170429.c and functions20170429.h.

**Comment by danielskatz** [comment]
@jmichel7 - back to you...

**Comment by jmichel7** [comment]
Most points I touched on have been improved. They are not yet perfect, but perhaps that's the end
of my comments.

**Comment by danielskatz** [comment]
@jmichel7 there are still a number of boxes unchecked.  Can you either check them, or explain what you think needs to be done before you are ready to check them?

**Comment by jmichel7** [comment]
I checked more boxes.
I did not find the test to verify functionality in the archive. Also there are still no community guidelines.
Is this last necessary?

**Comment by luk87** [comment]
@jmichel7 regarding the test I put a manual test in the documentation: the formula for the dimension of the tableaux [2,1] as function of the rank for the four algebras. Regarding the guidelines I assumed these were disciplined by the license and GitHub, but eventually I could add some more specific statements.

**Comment by danielskatz** [comment]
The idea of community guidelines is that the project should be welcoming of new contributors, and should explain how they can join in and contribute.  The text can be somewhat generic, but needs to be part of the project repo/documentation.

**Comment by luk87** [comment]
@danielskatz thank you for clarifying. I have added the guidelines in the README.

**Comment by danielskatz** [comment]
@jmichel7 please see if you are ok with the test and guidelines now

**Comment by jmichel7** [comment]
You are a better judge than me if the criteria are now met, but I have checked the remaining buttons on
the reviewer questions.

**Comment by danielskatz** [comment]
@jmichel7 thanks for your work on this

@arfon over to you for the next step

**Comment by arfon** [comment]
@luk87 - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by luk87** [comment]
@jmichel7  thank you. @arfon ok, I will do this and I will be back with the DOI.

**Comment by luk87** [comment]
@arfon here is the DOI

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.572276.svg)](https://doi.org/10.5281/zenodo.572276)

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.572276 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.572276" target="_blank">10.5281/zenodo.572276</a> is the archive.

**Comment by arfon** [comment]
@luk87 - as per the [author guidelines](http://joss.theoj.org/about#author_guidelines) your `paper.md` should only contain two sections (Summary/References). The other content you have here (Installation/Usage/Example) should be moved to either the repository README or some other docs.

In addition, please make sure you're citing the references in your bibtex file correctly as they won't be included otherwise. You can read how to do that [here](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)

Please let me know when you've done this and I'll process the submission.

**Comment by luk87** [comment]
@arfon I have improved the paper and the references. However my affiliation in the meanwhile has changed. For the moment I have no affiliation, while when I submitted the paper I was affiliated with the Leibniz universität hannover. Could this be a problem and what affiliation should I specify in the paper?

**Comment by arfon** [comment]
> Could this be a problem and what affiliation should I specify in the paper?

I think it's OK to keep the affiliation the same. Have you completed your changes?

**Comment by luk87** [comment]
@arfon Yes I have completed the changes.

**Comment by arfon** [comment]
@jmichel7 - many thanks for your review here and to @danielskatz for editing this one ✨ 

@luk87 - your paper is now accepted into JOSS and your DOI is http://dx.doi.org/10.21105/joss.00221 ⚡️ 🚀 💥
